src/Pos/Types/Types.hs

```diff
@@ -197,14 +197,23 @@ import           Pos.Util               (Color (Magenta), colorize)
 -- | Coin is the least possible unit of currency.
 newtype Coin = Coin
     { getCoin :: Word64
-    } deriving (Show, Ord, Eq, Bounded, Generic, Hashable, Data, NFData)
+    } deriving (Show, Ord, Eq, Generic, Hashable, Data, NFData)
 
 instance Buildable Coin where
     build (Coin n) = bprint (int%" coin(s)") n
 
+instance Bounded Coin where
+    minBound = Coin 0
+    maxBound = Coin maxCoinVal
+
+maxCoinVal :: Word64
+maxCoinVal = 44999999999999999
```

Comment:
We need to store total number of coins in system and in a long term it will become 45000000000000000, so I will change this line I think.